# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1383107619 14400
#      Wed Oct 30 00:33:39 2013 -0400
# Node ID f461e4a4768031166fbf8ff678e099d75752e46a
# Parent  0b81d65d3c49315f0e37a8e808d1ed36f28d01fc
* main.in.cc (main): Eliminate some #ifdefs.

diff --git a/src/main.in.cc b/src/main.in.cc
--- a/src/main.in.cc
+++ b/src/main.in.cc
@@ -394,73 +394,61 @@ strsave (const char *s)
 int
 main (int argc, char **argv)
 {
   int retval = 0;
 
 #if defined (HAVE_OCTAVE_GUI)
   bool start_gui = true;
   bool gui_libs = true;
+#else
+  bool start_gui = false;
+  bool gui_libs = false;
 #endif
 
   std::string octave_bindir = get_octave_bindir ();
 
-  std::string file = octave_bindir + dir_sep_char;
-
-#if defined (HAVE_OCTAVE_GUI)
-  file += "octave-gui";
-#else
-  file += "octave-cli";
-#endif
+  std::string file = octave_bindir + dir_sep_char
+    + (gui_libs ? "octave-gui" : "octave-cli");
 
   char **new_argv = new char * [argc + 1];
 
   int k = 0;
   new_argv[k++] = strsave ("octave");
 
   for (int i = 1; i < argc; i++)
     {
       if (! strcmp (argv[i], "--no-gui-libs"))
         {
           // Run the version of Octave that is not linked with any GUI
           // libraries.  It may not be possible to do plotting or any
           // ui* calls, but it will be a little faster to start and
           // require less memory.  Don't pass the --no-gui-libs option
           // on as that option is not recognized by Octave.
 
-#if defined (HAVE_OCTAVE_GUI)
           gui_libs = false;
-#endif
           file = octave_bindir + dir_sep_char + "octave-cli";
         }
       else if (! strcmp (argv[i], "--no-gui"))
         {
           // If we see this option, then we can just exec octave; we
           // don't have to create a child process and wait for it to
           // exit.  But do exec "octave", not "octave-cli", because even
           // if the --no-gui option is given, we may be asked to do some
           // plotting or ui* calls.
 
-#if defined (HAVE_OCTAVE_GUI)
           start_gui = false;
-#endif
           new_argv[k++] = argv[i];
         }
       else
         new_argv[k++] = argv[i];
     }
 
   new_argv[k] = 0;
 
-#if ! defined (HAVE_OCTAVE_GUI) || defined (__WIN32__) || defined (__CYGWIN__)
-
-  retval = octave_exec (file, new_argv);
-
-#else
-
   if (gui_libs && start_gui && have_controlling_terminal ())
     {
       install_signal_handlers ();
 
       gui_pid = fork ();
 
       if (gui_pid < 0)
         {
@@ -499,18 +487,16 @@ main (int argc, char **argv)
                   break;
                 }
             }
         }
     }
   else
     retval = octave_exec (file, new_argv);
 
-#endif
-
   return retval;
 }
 
 /*!
 @mainpage Source code documentation for GNU Octave
 
 GNU Octave is a high-level language, primarily intended for numerical
 computations.  It provides a convenient interactive command line
